packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -877,10 +886,16 @@ abstract class FlutterCommand extends Command<void> {
 
     String codeSizeDirectory;
     if (argParser.options.containsKey(FlutterOptions.kAnalyzeSize) && boolArg(FlutterOptions.kAnalyzeSize)) {
-      final Directory directory = globals.fsUtils.getUniqueDirectory(
+      Directory directory = globals.fsUtils.getUniqueDirectory(
         globals.fs.directory(getBuildDirectory()),
         'flutter_size',
       );
+      if (argParser.options.containsKey(FlutterOptions.kCodeSizeDirectory) && argResults[FlutterOptions.kCodeSizeDirectory] != null) {
```

Comment:
Does stringArg return null if nothing was passed? you could simplify this check a bit

Reply:
It can, but elsewhere we treat these like they don't return null.

Reply:
I changed this around a little ot match the style elsehwere